frontend/packages/component-library/src/common/contexts/ThemeContext.tsx

```diff
@@ -21,20 +24,46 @@ interface ThemeContextProps {
  */
 const ThemeContext = createContext<ThemeContextProps | undefined>(undefined);
 
-/** Provides code with values of ThemeContext inside of React component.
- * Please note that in order to work correctly - it requires component you're working in to be wrapped by ThemeProvider
- * anywhere on higher level
+/**
+ * Provides code with values of `ThemeContext` inside of a React component.
  *
- * Usage example:
+ * Default behavior: by default `isThemeOptional` is set to `false`, which means `useTheme` requires
+ * the component you're working in to be wrapped by `ThemeProvider` anywhere on a higher level.  If
+ * your component is not wrapped in a `ThemeProvider`, `useTheme` will throw an error.
+ *
+ * Usage example (default use):
  *    const {theme, toggleTheme} = useTheme();
+ *
+ * Advanced: setting `isThemeOptional` to `true` changes this default behavior and does not
```

Comment:
Nit, might be worth also adding a few words about when/why this is actually needed